### PR TITLE
Add sgi Previews

### PR DESCRIPTION
**Pull request by micbar**
## Description

This is adding support for SGI images
https://en.wikipedia.org/wiki/Silicon_Graphics_Image

The new supported file extensions are
- `.sgi` or `.rgb`
    3 colour channels
- `.rgba`
    3 colour cha-nnels and alpha
- `.bw` or `.int`
    black and white
- `.inta`
    black and white and alpha

These files will be associated with the mimetype `image/sgi`. 
This mimetype is not officially registered. https://www.iana.org/assignments/media-types/media-types.xhtml

### Dependency
- This needs the imagick php extension to be installed.
## Related Issue
https://github.com/owncloud/enterprise/issues/4131

## How Has This Been Tested?

- [x] Manual Test using demo files 
- [x] Manual Test in the details view
- [x] Manual Test with files_mediaviewer

### Demo Files
[RGB.zip](https://github.com/owncloud/core/files/5006697/RGB.zip)

### Screenshot File List
Screenshot_2020-07-31 RGB - Files - ownCloud

### Screenshot Details View
Screenshot_2020-07-31 Files - ownCloud

### Screenshot Mediaviewer
Screenshot_2020-07-31 RGB - Files - ownCloud(1)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [x] Documentation ticket raised: https://github.com/owncloud/docs/issues/2735
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: lib/private/PreviewManager.php [hunk 1]
```diff
@@ -268,7 +268,8 @@ protected function getEnabledDefaultProvider() {
 			'OC\Preview\GIF',
 			'OC\Preview\BMP',
 			'OC\Preview\Heic',
-			'OC\Preview\XBitmap'
+			'OC\Preview\XBitmap',
+			'OC\Preview\SGI',
```
Comment (micbar) [inline lib/private/PreviewManager.php:272]: This is questionable, could also be not default

**Comment by mmattel** [comment]
Maybe to add, `This format is supported by ImageMagick`, see Wikipedia (https://en.wikipedia.org/wiki/Silicon_Graphics_Image)

Please add a changelog 😅

File: lib/private/PreviewManager.php (hunk 1 above)
Comment (micbar) [reply lib/private/PreviewManager.php:272]: ok, leave it default

**Comment by micbar** [comment]
@mmattel Raised a docs ticket https://github.com/owncloud/docs/issues/2735

**Review by ChrisEdS — APPROVED** [review]
Works as expected. <3

**Comment by mmattel** [comment]
Will this go when merged into 10.5 or in a subsequent release ?

**Comment by micbar** [comment]
> Will this go when merged into 10.5 or in a subsequent release ?

10.5.0 is closed.

**Review by DeepDiver1975 — APPROVED** [review]

**Comment by micbar** [comment]
@DeepDiver1975 I am not sure about that mimetypes.

Changed it.

**Comment by micbar** [comment]
Reduced it again to the simple set.

**Review by jvillafanez — APPROVED** [review]

**Comment by mmattel** [comment]
Are there tests needed ? or merging without (already approved...)

**Comment by micbar** [comment]
basic unit tests are already done for the parent class

**Comment by mmattel** [comment]
Maybe asking @phil-davis for help?
I can not assists, tests are my weak point, sorry.

**Comment by micbar** [comment]
I can also write tests 😄 , IMO the existing tests are enough.

This pr is ready to merge

**Comment by phil-davis** [comment]
We do not currently have specific automated end-to-end acceptance tests for preview generation of each of the supported file types. There are just "generic" test scenarios for "ordinary" text files.

Making such tests in a useful way (i.e. that know what preview result to really expect for each file type) would need to be a separate issue that can be done according to priority.

**Comment by micbar** [comment]
ok, merging now.

**Pull request merged**